cmd/expiration-mailer/main.go

```diff
@@ -302,15 +302,7 @@ func (m *mailer) findExpiringCertificates() error {
 		var certs []core.Certificate
 		for _, serial := range serials {
 			var cert core.Certificate
-			err := m.dbMap.SelectOne(&cert,
-				`SELECT
-				cert.*
-				FROM certificates AS cert
-				WHERE serial = :serial`,
-				map[string]interface{}{
-					"serial": serial,
-				},
-			)
+			cert, err := sa.SelectCertificate(m.dbMap, "WHERE serial = ?", serial)
```

Comment:
This is a nice refactor; can you split it, and the test refactor, into their own PR?